app/models/concerns/reviewable_action_builder.rb

```diff
@@ -123,28 +166,70 @@ def build_action(
   end
 
   def perform_no_action_user(performed_by, args)
-    create_result(:success, :approved)
+    create_result(:success, :rejected)
   end
 
-  def perform_silence_user(performed_by, args)
+  def perform_no_action_post(performed_by, args)
     create_result(:success, :rejected)
   end
 
+  def perform_silence_user(performed_by, args)
+    create_result(:success, :approved)
+  end
+
   def perform_suspend_user(performed_by, args)
-    create_result(:success, :rejected)
+    create_result(:success, :approved)
   end
 
   def perform_delete_user(performed_by, args, &)
     delete_user(target_user, delete_opts, performed_by) if target_user
-    create_result(:success, :rejected, [], recalculate_score: false, &)
+    create_result(:success, :approved, [], false, &)
   end
 
   def perform_delete_and_block_user(performed_by, args, &)
     delete_options = delete_opts
-    delete_options.merge!(block_email: true, block_ip: true) if Rails.env.production?
+    if Rails.env.production?
+      delete_options = delete_options.merge(block_email: true, block_ip: true)
+    end
 
     delete_user(target_user, delete_options, performed_by) if target_user
-    create_result(:success, :rejected, [], recalculate_score: false, &)
+    create_result(:success, :approved, [], false, &)
+  end
+
+  def perform_delete_post(performed_by, _args)
+    PostDestroyer.new(performed_by, target_post, reviewable: self).destroy
+    create_result(:success, :approved, [created_by_id], false)
+  end
+
+  def perform_hide_post(performed_by, _args)
+    target_post.hide!(PostActionType.types[:inappropriate])
```

Comment:
Is `inappropriate` the only reason for hiding?

Reply:
Oh, good catch. `ReviewableFlaggedPost` has some extra magic for handling flags that I didn't get back to porting across. Let me have a look and see what I can do. 🙂

Reply:
Porting that is more complicated than I'd like, so I'll leave that for a follow-up PR.